vaccine_feed_ingest/runners/wa/state/normalize.py

```diff
@@ -0,0 +1,194 @@
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+logger = getLogger(__file__)
+
+
+def _get_id(site: dict) -> str:
+    # The graphql docs say this id should be unique to this location within
+    # this data source
+    locationId = site["locationId"]
+
+    return locationId
+
+
+def _get_contacts(site: dict) -> List[schema.Contact]:
+    # From the docs for this data source:
+    # These are nullable fields, but one is "guaranteed" to be non nullable
+    phone, email, scheduling_link = (
+        site.get("phone"),
+        site.get("email"),
+        site.get("schedulingLink"),
+    )
+
+    info_link = site.get("infoLink")
+    try:
+        phone_contact = (
+            schema.Contact(
+                contact_type="booking",
+                phone=phone,
+            )
+            if phone is not None
+            else None
+        )
+    except ValidationError:
+        logger.warning(f"Invalid phone contact {phone}")
+        phone_contact = None
+
+    try:
+        email_contact = (
+            schema.Contact(
+                contact_type="booking",
+                email=email,
+            )
+            if email is not None
+            else None
+        )
+    except ValidationError:
+        logger.warning(f"Invalid email contact {email}")
+        email_contact = None
+
+    try:
+        website_contact = (
+            schema.Contact(
+                contact_type="booking",
+                website=scheduling_link,
+            )
+            if scheduling_link is not None
+            else None
+        )
+    except ValidationError:
+        logger.warning(f"Invalid scheduling_link contact {scheduling_link}")
+        website_contact = None
+
+    try:
+        general_contact = (
+            schema.Contact(contact_type="general", website=info_link)
+            if info_link
+            else None
+        )
+    except ValidationError:
+        logger.warning(f"Invalid info_link contact {info_link}")
+        general_contact = None
+
+    return [
+        contact
+        for contact in (phone_contact, email_contact, website_contact, general_contact)
+        if contact is not None
+    ]
+
+
+def _get_wheelchair(site):
+    if site.get("wheelchairAccessible") is True:
+        return "yes"
+    if site.get("wheelchairAccessible") is False:
+        return "no"
+
+    return None
+
+
+def _get_supply_level(site):
+    if site["vaccineAvailability"] == "AVAILABLE":
+        return "in_stock"
+
+    if site["vaccineAvailability"] == "UNAVAILABLE":
+        return "out_of_stock"
+
+    # Docs say the value should be UNKNOWN
+    return None
+
+
+def _get_vaccine_type(type_string):
+    if type_string == "johnsonAndJohnson":
+        return "johnson_johnson_janssen"
+    if type_string == "moderna":
+        return "moderna"
+    if type_string == "pfizer":
+        return "pfizer_biontech"
+    return None
+
+
+def _is_good_zip(zip):
+    if len(zip) != 5:
```

Comment:
You can probably grab the first 5 digits of the zip if there are more?

Reply:
The weird data I saw had chars at the front actually